debug/captions.html

```diff
@@ -253,87 +271,124 @@ <h3>footer</h3>
 
     m2ts.addEventListener('change', function() {
       var reader = new FileReader();
+      var remuxedSegments = [];
+      var parsedCaptions = [];
+      var remuxedInitSegment = null;
+      var remuxedBytesLength = 0;
+      var createInitSegment = false;
+      var bytes;
+
+      var combined = document.querySelector('#combined-output').checked;
+      var outputType = document.querySelector('input[name="output"]:checked').value;
+      var resetTransmuxer = $('#reset-tranmsuxer').checked;
+      var partial = $('#partial-transmuxer').checked;
 
       // do nothing if no file was chosen
       if (!this.files[0]) {
         return;
       }
 
-      reader.addEventListener('loadend', function() {
+      var onData = function(event) {
+        if (event.type === outputType) {
+          remuxedSegments.push(event);
+
+          if (!partial) {
+            remuxedBytesLength += event.data.byteLength;
+            remuxedInitSegment = event.initSegment;
+          } else {
+            remuxedBytesLength += event.data.boxes.byteLength;
+            remuxedInitSegment = event.data.track.initSegment;
+          }
+        }
+      };
+
+      var onDone = function (source) {
+        var offset = 0;
+
+        if (source && !source.toLowerCase().includes(outputType)) {
+          return;
+        }
+
+        console.log('done');
+
+        if (createInitSegment) {
+          bytes = new Uint8Array(remuxedInitSegment.byteLength + remuxedBytesLength)
+          bytes.set(remuxedInitSegment, offset);
+          offset += remuxedInitSegment.byteLength;
+          createInitSegment = false;
+        } else {
+          bytes = new Uint8Array(remuxedBytesLength);
+        }
+
+        var i;
```

Comment:
I don't think this is needed.

Reply:
oh, true the one in the loop should be sufficient